topic_monitor/README.md

```diff
@@ -0,0 +1,86 @@
+# Visualizing the effector of QoS policies
+
+This demo uses a “topic monitor” that can be used to visualize the statistics of ROS 2 topics that are publishing sequential data.
+
+## Background
+Please read the [About Quality of Service Settings](https://github.com/ros2/ros2/wiki/About-Quality-of-Service-Settings) page for background information about the Quality of Service settings available in ROS 2.
+
+## Running the demo
+To visualize the reception rate, we will use a “topic monitor” that periodically calculates the reception rate of topics with publishers of periodic, sequential data.
+
+If you have the Python3 `matplotlib` package installed, you can use the `--display` option to plot the reception rate of topics:
+```
+topic_monitor --display
+```
+
+Alternatively, if you have ROS 1 installed, you can use the ROS 1 - ROS 2 bridge to plot the reception rate using ROS 1 tools such as `rqt`, or log it using `rosbag`.
+Be sure to run the bridge with `--bridge-all-2to1-topics` so that all topics will be bridged, that way `rqt` will be able to list the topics before it has subscribed to them.
+
+**For all invocations that follow make sure the same `ROS_DOMAIN_ID` has been set to the same value on the respective machines.**
+
+
+### Comparing reliability QoS settings
+You will need two machines with ROS 2: one mobile and one stationary.
+
+1. Run the `topic_monitor_launch_reliability_demo` executable on the stationary machine.
+This will start two nodes: one publishing in “reliable” mode, and one in “best effort”.
+1. Start the monitor on a mobile machine such as a laptop.
+Use `topic_monitor --display --allowed-latency 5` to account for any latency that may occur re-sending the reliable messages.
+1. Take the mobile machine out of range of the monitor, and observe how the reception rates differ for the different topics.
+
+You should see that the "reliable" topic has a reception rate that is almost always either 0 or 100%, while the "best effort" topic has a reception rate that fluctuates based on the strength of the connection.
```

Comment:
I am a bit puzzled why this is the case. I would have expected reliable data to arrive (even if with a large latency because of repeated retries) until the connection is completely gone.